compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenInterfaces.td

```diff
@@ -549,6 +549,29 @@ def IREECodegen_UKernelProviderInterface :
         return failure();
       }]
     >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Creates and replaces the |contextual_op| with a ukernel referenced by
+        |name| backed by this attribute. Takes |target_configuration| in
+        case the provider wants to make any feature specific decisions for
+        which implementation to provide. It takes in |inputs|, |outputs| and
+        |other_operands| and adds to it a few context-specific as well as
+        backend specific implementations to create the ukernel op.
+      }],
+      /*retTy=*/"::mlir::LogicalResult",
+      /*methodName=*/"createAndReplaceWithUkernelOp",
+      /*args=*/(ins "::mlir::RewriterBase&":$b,
+                    "::mlir::StringRef":$name,
```

Comment:
Instead of `name` we can pass the `ukernel_descriptor` attribute which includes the ukernel kind. That way we can make the default implementation the same as the default replacement helpers implemented in the `Lower*UKernelPass`es.

Reply:
Oh darn, I didn't think about the dependency issue. Also I just realized that'd introduce a tensor and memref dep in the Codegen dialect so it's not a good idea. Sorry for the bad suggestion. What we can try instead is making the return type `std::optional<LogicalResult>` to allow for falling back to the default implementation by returning `std::nullopt`.

For `UKernelDescriptorAttr`, in theory the provider can infer the kind from the name so that's maybe ok to leave out for now. If we find it needed in the future we can work out the implementation then.

Reply:
LGTM, can you add a comment to the method description describing the return modes? Something like

```
Returning std::nullopt indicates falling back to default implementation. Otherwise success/failure indicates the status of the custom replacement implementation.
```